mongodbatlas/provider_test.go

```diff
@@ -30,6 +35,14 @@ func init() {
 			return muxProviderFactoryUsingExistingSdkV2(testAccProviderSdkV2)(), nil
 		},
 	}
+
+	config := Config{
+		PublicKey:    os.Getenv("MONGODB_ATLAS_PUBLIC_KEY"),
+		PrivateKey:   os.Getenv("MONGODB_ATLAS_PRIVATE_KEY"),
+		BaseURL:      os.Getenv("MONGODB_ATLAS_BASE_URL"),
+		RealmBaseURL: os.Getenv("MONGODB_REALM_BASE_URL"),
+	}
+	testMongoDBClient, _ = config.NewClient(context.Background())
```

Comment:
That is still used in existing un-migrated resource tests right?

Reply:
Correct. Essentially we can leave either `testMongoDBClient`, or the usage of `testAccProviderSdkV2.Meta().(*MongoDBClient)`. Between both I prefer testMongoDBClient.

Reply:
I don't think we should change that in all the other resource and data source tests unless required as we'd be running all the tests then. At least not as part of this PR. We can address that separately.

Reply:
Thats okay, in that case I would however remove the created `testMongoDBClient` in favor of the existing `testAccProviderSdkV2.Meta().(*MongoDBClient)` that is being used in acceptance tests, mainly for consistency.

Reply:
While that would be consistent in the sense that we use only `testAccProviderSdkV2.Meta().(*MongoDBClient)` throughout our acceptance tests to get the client, it would be inconsistent in context of a **Framework-based migrated** resource and it's tests to continue using an SDKv2 based entity. This would also require us to update all the tests once we are ready to get rid of everything that SDKv2 in our provider.

We can instead rename `testMongoDBClient` to  something like `testMongoDBClient_Fw` (open to suggestions for the name) to clarify that this is specific to the Framework migrated resources.
Let me know what you think.

Reply:
If we will progressively transition from testAccProviderSdkV2.Meta().(*MongoDBClient) to using the new testMongoDBClient then I think the existing comments are sufficient